File: generate_xlsx.py
import pandas as pd
import numpy as np
import random
from datetime import datetime, timedelta

# Configuration
num_rows = 5000
clients = ["GlobalCorp", "TechNova", "Starlight Media", "Alpha Logistics", "BlueSky Retail", "Nexus Health", "Omega Energy", "Peak Finance"]
regions = ["North America", "EMEA", "APAC", "LATAM"]
categories = ["Software", "Hardware", "Consulting", "Support"]
services = ["SaaS Subscription", "Implementation", "Maintenance", "Strategy Audit", "Cloud Hosting"]

def random_date(start, end):
    return start + timedelta(days=random.randint(0, (end - start).days))

start_date = datetime(2020, 1, 1)
end_date = datetime(2026, 12, 31)

data = []
for i in range(num_rows):
    date = random_date(start_date, end_date).strftime('%Y-%m-%d')
    unit_price = round(random.uniform(50, 550), 2)
    quantity = random.randint(1, 100)
    gross_revenue = round(unit_price * quantity, 2)
    discount_rate = round(random.uniform(0, 0.15), 4)
    net_revenue = round(gross_revenue * (1 - discount_rate), 2)
    cogs_percent = round(random.uniform(0.3, 0.5), 4)
    cogs_amount = round(net_revenue * cogs_percent, 2)
    marketing_spend = round(net_revenue * random.uniform(0.05, 0.15), 2)
    fixed_costs = round(random.uniform(500, 700), 2)
    op_expenses = round(marketing_spend + fixed_costs, 2)
    ebitda = round(net_revenue - cogs_amount - op_expenses, 2)
    ebitda_margin = round(ebitda / net_revenue, 4) if net_revenue != 0 else 0
    tax_rate = 0.21
    net_profit = round(ebitda * (1 - tax_rate), 2)

    data.append({
        "Date": date,
        "Transaction ID": f"TXN-{1000 + i}",
        "Client Name": random.choice(clients),
        "Region": random.choice(regions),
        "Product Category": random.choice(categories),
        "Service Line": random.choice(services),
        "Unit Price": unit_price,
        "Quantity": quantity,
        "Gross Revenue": gross_revenue,
        "Discount Rate": f"{round(discount_rate * 100, 2)}%",
        "Net Revenue": net_revenue,
        "COGS Percent": f"{round(cogs_percent * 100, 2)}%",
        "COGS Amount": cogs_amount,
        "Marketing Spend": marketing_spend,
        "Fixed Costs": fixed_costs,
        "Operating Expenses": op_expenses,
        "EBITDA": ebitda,
        "EBITDA Margin": f"{round(ebitda_margin * 100, 2)}%",
        "Tax Rate": "21%",
        "Net Profit": net_profit
    })

df = pd.DataFrame(data)
df.to_excel("financial_analysis_5000_rows.xlsx", index=False, engine='openpyxl')
print("File generated: financial_analysis_5000_rows.xlsx")

File: generate_xlsx.js
import XLSX from 'xlsx';
import fs from 'fs';

const headers = [
    "Date", "Transaction_ID", "Client_Name", "Region", "Product_Category", 
    "Service_Line", "Unit_Price", "Quantity", "Gross_Revenue", "Discount_Rate", 
    "Net_Revenue", "COGS_Percent", "COGS_Amount", "Marketing_Spend", "Fixed_Costs", 
    "Operating_Expenses", "EBITDA", "EBITDA_Margin", "Tax_Rate", "Net_Profit"
];

const clients = ["GlobalCorp", "TechNova", "Starlight Media", "Alpha Logistics", "BlueSky Retail", "Nexus Health", "Omega Energy", "Peak Finance"];
const regions = ["North America", "EMEA", "APAC", "LATAM"];
const categories = ["Software", "Hardware", "Consulting", "Support"];
const services = ["SaaS Subscription", "Implementation", "Maintenance", "Strategy Audit", "Cloud Hosting"];

const rows = [];

for (let i = 0; i < 100; i++) {
    const date = new Date(2020 + Math.floor(Math.random() * 7), Math.floor(Math.random() * 12), Math.floor(Math.random() * 28) + 1);
    const unitPrice = parseFloat((Math.random() * 500 + 50).toFixed(2));
    const quantity = Math.floor(Math.random() * 100) + 1;
    const grossRevenue = unitPrice * quantity;
    const discountRate = parseFloat((Math.random() * 0.15).toFixed(4));
    const netRevenue = grossRevenue * (1 - discountRate);
    const cogsPercent = parseFloat((Math.random() * 0.2 + 0.3).toFixed(4));
    const cogsAmount = netRevenue * cogsPercent;
    const marketingSpend = netRevenue * parseFloat((Math.random() * 0.1 + 0.05).toFixed(4));
    const fixedCosts = 500 + Math.random() * 200;
    const opExpenses = marketingSpend + fixedCosts;
    const ebitda = netRevenue - cogsAmount - opExpenses;
    const ebitdaMargin = ebitda / netRevenue;
    const taxRate = 0.21;
    const netProfit = ebitda * (1 - taxRate);

    rows.push([
        date.toISOString().split('T')[0],
        `TXN-${1000 + i}`,
        clients[Math.floor(Math.random() * clients.length)],
        regions[Math.floor(Math.random() * regions.length)],
        categories[Math.floor(Math.random() * categories.length)],
        services[Math.floor(Math.random() * services.length)],
        unitPrice,
        quantity,
        parseFloat(grossRevenue.toFixed(2)),
        parseFloat((discountRate * 100).toFixed(2)) + "%",
        parseFloat(netRevenue.toFixed(2)),
        parseFloat((cogsPercent * 100).toFixed(2)) + "%",
        parseFloat(cogsAmount.toFixed(2)),
        parseFloat(marketingSpend.toFixed(2)),
        parseFloat(fixedCosts.toFixed(2)),
        parseFloat(opExpenses.toFixed(2)),
        parseFloat(ebitda.toFixed(2)),
        parseFloat((ebitdaMargin * 100).toFixed(2)) + "%",
        "21%",
        parseFloat(netProfit.toFixed(2))
    ]);
}

const wb = XLSX.utils.book_new();
const ws = XLSX.utils.aoa_to_sheet([headers, ...rows]);
XLSX.utils.book_append_sheet(wb, ws, "Financial_Analysis");

const buf = XLSX.write(wb, { type: 'buffer', bookType: 'xlsx' });
fs.writeFileSync("financial_analysis_2020_2026.xlsx", buf);
console.log("File generated: financial_analysis_2020_2026.xlsx");
